erpnext/crm/doctype/lead/lead.py

```diff
@@ -123,12 +129,17 @@ def set_lead_name(self):
 		if not self.lead_name:
 			# Check for leads being created through data import
 			if not self.company_name and not self.email_id and not self.flags.ignore_mandatory:
-				frappe.throw(_("A Lead requires either a person's name or an organization's name"))
+				frappe.throw(_("This Lead requires either a person's name or an organization's name"))
 			elif self.company_name:
 				self.lead_name = self.company_name
 			else:
 				self.lead_name = self.email_id.split("@")[0]
 
+	def set_company_name(self):
+		if not self.company_name:
+			if not self.flags.ignore_mandatory:
+				frappe.throw(_("This Lead requires an organization's name"))
```

Comment:
This could better be achieved with the "Mandatory Depends On" field of `company_name`